### ISSUE TITLE
Bifrost should advertise MUC support on it's domain

### ISSUE THREAD

**Issue by Half-Shot** [matrix-org/matrix-bifrost#131]
So that clients aren't tuned away when they try to poll "matrix.org" for MUCs.

**Comment by ge0rg** [comment]
I've dug into Gajim / nbxmpp source code and found out the following.

1. The Gajim error message "Address does not belong to a group chat server" is misleading, it is actually querying the room and not the domain JID.
2. Gajim is [looking for a conference identity](https://dev.gajim.org/gajim/python-nbxmpp/-/blob/master/nbxmpp/structs.py#L228), i.e. `<identity category="conference" type=... />` and for the MUC feature `<feature var='http://jabber.org/protocol/muc'/>` - as Bifröst is only offering a gateway identity, this check fails and Gajim refuses to join.

Your short-term solution is probably to add an `<identity category="conference" type=."text" />` to **at least** the room's disco#info, and **maybe also** to the domain's.

I've [raised this question on the XMPP standards ML](https://mail.jabber.org/pipermail/standards/2020-October/037784.html), so let's wait and see how it turns out.

**Comment by horazont** [comment]
Relevant: https://github.com/matrix-org/matrix-bifrost/issues/48

**Comment by Half-Shot** [comment]
Strange, because we do already set that:

```
<iq xmlns="jabber:client" to="matrix.org" type="get" id="918f261e-290d-4485-8bf1-2aec2419538f">
  <query xmlns="http://jabber.org/protocol/disco#info" />
</iq>
```

```
<iq xmlns="jabber:client" to="me" from="matrix.org" type="result" id="918f261e-290d-4485-8bf1-2aec2419538f">
  <query xmlns="http://jabber.org/protocol/disco#info">
    <identity type="text" category="conference" name="Bifrost" />
    <feature var="http://jabber.org/protocol/disco#info" />
    <feature var="http://jabber.org/protocol/disco#items" />
    <feature var="http://jabber.org/protocol/muc" />
    <feature var="jabber:iq:version" />
    <feature var="jabber:iq:search" />
</query>
</iq>
```

Which feels correct?

**Comment by Half-Shot** [comment]
Our room identity is wrong though (not sure why we did this?):

```js
            discoInfo.identity.add({
                category: "gateway",
                name: alias,
                type: "matrix",
            });
```

**Comment by ge0rg** [comment]
Yup, the response from a room is this: 

```xml
<iq id="6ae262fc92374ddcaf0ff56f55c37d3f" type="result" xml:lang="en" from="#bifrost#half-shot.uk@matrix.org" to="you">
  <query xmlns="http://jabber.org/protocol/disco#info">
    <identity name="#bifrost:half-shot.uk" type="matrix" category="gateway"/>
    <feature var="http://jabber.org/protocol/disco#info"/>
    <feature var="http://jabber.org/protocol/muc"/>
  </query>
</iq>
```

**Comment by Half-Shot** [comment]
Yep, should be fixed by #138
